From 406964b59113109d3f7bef2681aaf9362b10d661 Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Sun, 12 May 2013 18:24:57 +0300
Subject: [PATCH] resolvconf: enforce configuration defaults to avoid conflict

dhcpcd, for example, uses interface_order environment variable
which creates a conflict with resolv conf setting, as it takes
that value as default configuration.

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 resolvconf.in | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/resolvconf.in b/resolvconf.in
index 5015222..82b8b55 100644
--- a/resolvconf.in
+++ b/resolvconf.in
@@ -28,6 +28,13 @@ RESOLVCONF="$0"
 SYSCONFDIR=@SYSCONFDIR@
 LIBEXECDIR=@LIBEXECDIR@
 VARDIR=@VARDIR@
+
+# set configuration defaults
+dynamic_order="tap[0-9]* tun[0-9]* vpn vpn[0-9]* ppp[0-9]* ippp[0-9]*"
+interface_order="lo lo[0-9]*"
+name_server_blacklist="0.0.0.0"
+state_dir=""
+
 # Support original resolvconf configuration layout
 # as well as the openresolv config file
 if [ -f "$SYSCONFDIR"/resolvconf.conf ]; then
@@ -43,10 +50,6 @@ IFACEDIR="$VARDIR/interfaces"
 METRICDIR="$VARDIR/metrics"
 PRIVATEDIR="$VARDIR/private"
 
-: ${dynamic_order:=tap[0-9]* tun[0-9]* vpn vpn[0-9]* ppp[0-9]* ippp[0-9]*}
-: ${interface_order:=lo lo[0-9]*}
-: ${name_server_blacklist:=0.0.0.0}
-
 error_exit()
 {
 	echo "$*" >&2
-- 
1.8.1.5

